url294e6c45f3_Ecdosis_NMergeNew_tgz-pJ8-TestJ8/src/edu/luc/nmerge/mvd/MVDFile.java

```diff
@@ -0,0 +1,771 @@
+			:file;
+		description = description.replace("\"", "\\\"");
+		if ( username == null )
+			username = "root";
+		if ( password == null )
+			password = "jabberw0cky";
+		if ( url == null )
+			url = "jdbc:mysql://localhost:3306/";
+		if ( jdbcClass == null )
+			jdbcClass = "com.mysql.jdbc.Driver";
+		if ( dbName == null )
+			dbName = "dv";
+		if ( nameField == null )
+			nameField = "name";
+		if ( descField == null )
+			descField = "description";
+		if ( dataField == null )
+			dataField = "data";
+		if ( tableName == null )
+			tableName = "works";
+		if ( folderField == null )
+			folderField = "folder_id";
+		try
+        {
+            Class.forName(jdbcClass).newInstance();
+            conn = DriverManager.getConnection(url+dbName, username, password);
+            if ( conn != null )
+            {
+	            stmt = conn.createStatement();
+	            if ( stmt != null )
+	            {
+	            	String query;
+	            	rs = stmt.executeQuery( "select * from "+dbName+"."+tableName
+	            		+" where "+nameField+"=\""+name+"\"");
+	            	if ( rs.first() )
+	            	{
+	            		query = "update "+dbName+"."+tableName
+	            			+" set "+dbName+"."+tableName+"."+descField+"=\""
+	            			+description+"\","+dbName+"."+tableName+"."+dataField+"=\""
+	            			+data+"\","+folderField+"=\""+folderId+"\" where "
+	            			+nameField+"=\""+name+"\"";
+	            	}
+	            	else
+	            	{
+	            		query = "insert into "+dbName+"."+tableName+" ("+nameField+","
+	            			+descField+","+dataField+","+folderField+") values(\""
+	            			+name+"\",\""+description+"\",\""+data+"\",\""+folderId+"\");";
+	            	}
+	            	stmt = conn.createStatement();
+	            	stmt.executeUpdate( query );
+	            }
+	            conn.close();
+            }
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+        finally
+        {
+           	if (rs != null) {
+                try {
+                    rs.close();
+                } catch (SQLException sqlEx) { } // ignore
+
+                rs = null;
+            }
+            if (stmt != null) {
+                try {
+                    stmt.close();
+                } catch (SQLException sqlEx) { } // ignore
+
+                stmt = null;
+            if (conn != null)
+            {
+                 }
+                try
+                {
+                    conn.close ();
+                }
+                catch (Exception e) { /* ignore close errors */ }
+            }
+        }
+	}
+	/**
+	 * Write to an actual file
+	 * @param dst the location of the file
+	 * @param str the string of data to write
+	 * @throws Exception
+	 */
+	private static void writeToFile( File dst, String str ) throws Exception
+	{
+		if ( dst.exists() )
+			dst.delete();
+		dst.createNewFile();
+		FileWriter fw = new FileWriter( dst );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced at line 213. The resource is closed at line 214. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *fw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).